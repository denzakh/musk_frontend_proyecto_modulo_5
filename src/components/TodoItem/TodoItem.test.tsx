import { render, screen } from '@testing-library/react';
import { TodoItem } from './index';

const item = {
    id: '1',
    title: 'title1',
    text: 'Aprender Jest',
    completed: false,
};

const testFn = () => {};

test('Muestra la tarea correctamente', () => {
    render(
        <TodoItem
            item={item}
            delItem={testFn}
            setCompletedItem={testFn}
            setEditedItem={testFn}
        />
    );
    const taskElement = screen.getByText('Aprender Jest');
    expect(taskElement).toBeInTheDocument();
});

test('Genera un snapshot del componente TodoItem', () => {
    const { asFragment } = render(
        <TodoItem
            item={item}
            delItem={testFn}
            setCompletedItem={testFn}
            setEditedItem={testFn}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});
