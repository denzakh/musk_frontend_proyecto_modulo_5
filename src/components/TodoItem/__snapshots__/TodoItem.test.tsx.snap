// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genera un snapshot del componente TodoItem 1`] = `
<DocumentFragment>
  <li
    class="li"
    data-testid="item1"
    id="1"
    title="item"
  >
    <input
      class="input"
      data-testid="checkboxItem1"
      type="checkbox"
    />
    <div
      class="textRow"
    >
      <strong
        class="title"
      >
        title1
      </strong>
      <div
        class="text"
      >
        Aprender Jest
      </div>
    </div>
    <button
      data-testid="editItem1"
      title="Edit"
      type="button"
    >
      ðŸ–‰
    </button>
    <button
      data-testid="deleteItem1"
      title="Delete"
      type="button"
    >
      ðŸ—‘
    </button>
  </li>
</DocumentFragment>
`;
